### PR TITLE
Merge use implicit booleaness like checks

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

First step before #6870 and following #8592, just merging without modifying the messages or optimizing anything for now to make it more reviewable and validating the premises.

Refs #6871 #6870

### CODE REVIEW

File: doc/whatsnew/fragments/6871.user_action [hunk 1]
```diff
@@ -0,0 +1,14 @@
+* The compare to empty string checker (``pylint.extensions.emptystring``) and the compare to
+  zero checker (``pylint.extensions.compare-to-zero``) have been removed and their checks are
+  now part of the implicit booleaness checker:
+
+  - ``compare-to-zero`` was renamed ``use-implicit-booleaness-not-comparison-to-zero``
+    and ``compare-to-empty-string`` was renamed ``use-implicit-booleaness-not-comparison-to-string``
+    and they now need to be enabled explicitly.
+  - The `pylint.extensions.emptystring`` and ``pylint.extensions.compare-to-zero`` extensions no longer exists and
+    needs to be removed from the ``load-plugins`` option.
```
Comment (Pierre-Sassoulas) [inline doc/whatsnew/fragments/6871.user_action:9]: Really considering to make an update tool for configuration files, it's not a lot of work and will permit to make breaking changes with more serenity.

File: pylint/checkers/refactoring/implicit_booleaness_checker.py [hunk 2]
```diff
@@ -70,6 +80,21 @@ class ImplicitBooleanessChecker(checkers.BaseChecker):
             "used to check for emptiness; Use implicit booleaness instead "
             "of a collection classes; empty collections are considered as false",
         ),
+        "C1804": (
+            '"%s" can be simplified to "%s" as an empty string is falsey',
+            "use-implicit-booleaness-not-comparison-to-string",
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/refactoring/implicit_booleaness_checker.py:85]:
```suggestion
            "consider-implicit-booleaness-str",
```

File: doc/data/messages/u/use-implicit-booleaness-not-comparison-to-zero/good.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+
```
Comment (DanielNoord) [inline doc/data/messages/u/use-implicit-booleaness-not-comparison-to-zero/good.py:1]:
```suggestion
```

Can we run `black` over these?

**Review by DanielNoord — DISMISSED** [review]

File: doc/data/messages/u/use-implicit-booleaness-not-comparison-to-zero/good.py (hunk 3 above)
Comment (Pierre-Sassoulas) [reply doc/data/messages/u/use-implicit-booleaness-not-comparison-to-zero/good.py:1]: Not on all of them but yeah, we're running ruff already. I'm going to open another MR.

Comment (Pierre-Sassoulas) [reply doc/data/messages/u/use-implicit-booleaness-not-comparison-to-zero/good.py:1]: #8650

**Review by DanielNoord — APPROVED** [review]

**Pull request merged**